staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -18,33 +18,58 @@ package cache
 
 import (
 	"fmt"
+	"math"
 	"sync"
 	"testing"
 	"time"
 
 	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/clock"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 	fcache "k8s.io/client-go/tools/cache/testing"
 )
 
+const timeFmt = "15:04:05.999999999"
+
 type testListener struct {
-	lock              sync.RWMutex
-	resyncPeriod      time.Duration
-	expectedItemNames sets.String
-	receivedItemNames []string
-	name              string
+	t                *testing.T
+	name             string
+	clock            clock.PassiveClock
+	resyncPeriod     time.Duration
+	checkTiming      bool // whether handle() checks timing
+	expectedItemKeys sets.String
+	lock             sync.RWMutex
```

Comment:
@MikeSpreitzer could just embed the mutex into the struct and do:

```
type testListener struct {
    sync.RWMutex
    <other stuff>
```

and then below do `l.Lock()`

Reply:
Yeah, I thought I would leave the original pattern in place.